http/jsonapi/request.go

```diff
@@ -434,6 +445,48 @@ func unmarshalAttribute(
 	return
 }
 
+func handleDecimal(attribute interface{}) (reflect.Value, error) {
+	var dec decimal.Decimal
+	var err error
+
+	switch v := attribute.(type) {
+	case string:
+		dec, err = decimal.NewFromString(v)
+		if err != nil {
+			return reflect.Value{}, err
+		}
+	case int:
+		dec = decimal.NewFromInt(int64(v))
+	case int8:
+		dec = decimal.NewFromInt(int64(v))
+	case int16:
+		dec = decimal.NewFromInt(int64(v))
+	case int32:
+		dec = decimal.NewFromInt(int64(v))
+	case int64:
+		dec = decimal.NewFromInt(int64(v))
+	case uint:
+		dec = decimal.NewFromInt(int64(v))
+	case uint8:
+		dec = decimal.NewFromInt(int64(v))
+	case uint16:
+		dec = decimal.NewFromInt(int64(v))
+	case uint32:
+		dec = decimal.NewFromInt(int64(v))
+	case uint64:
+		dec = decimal.NewFromInt(int64(v))
```

Comment:
Wouldn't it be better to format uints to string instead of casting to int64? At least uint64 could overflow here.